### PR TITLE
chore: integrate cocogitto for release

### LINKED ISSUE #17: Add changelog

### PR DESCRIPTION
**Pull request by JacquesDurand**
Add [cocogitto](https://github.com/cocogitto/cocogitto), a tool that:

- checks for conventional commit compliance
- bumps versions automatically based on semver + conventional commit
- generates changelog accordingly

closes #17

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,6 @@ on:
   schedule:
     # Every Sunday at 02:00
     - cron: 0 2 * * 0
-  workflow_dispatch: ~
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]: We should keep it, it allows us to execute the CI manually

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,6 +25,20 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+      - run: |
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]: Why do you need to configure the git user just to check the commits on the PR?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Generate Changelog
+        run: cog changelog --at ${{ steps.release.outputs.version }} -t full_hash > GITHUB_CHANGELOG.md
+      - name: Upload github release
+        uses: softprops/action-gh-release@v1
+        with:
+          body_path: GITHUB_CHANGELOG.md
+          tag_name: ${{ steps.release.outputs.version }}
+          
+  build:
+  name: Build
```
Comment (vincentchalamon) [inline .github/workflows/release.yaml]: Invalid indent

Comment (vincentchalamon) [reply .github/workflows/release.yaml]: Also, please rebase to get the code from https://github.com/JacquesDurand/php-documentation-generator/pull/37

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git config --global user.email github-actions@github.com
+          git config --global user.name github-actions
+      - name: Cocogitto release
```
Comment (vincentchalamon) [inline .github/workflows/release.yaml:18]: A release is already created on tag here: https://github.com/JacquesDurand/php-documentation-generator/pull/37

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        id: release
+        uses: oknozor/cocogitto-action@v3.4
+        with:
+          release: true
+          git-user: 'github-actions'
+          git-user-email: 'github-actions@github.com'
+      - name: Generate Changelog
```
Comment (vincentchalamon) [inline .github/workflows/release.yaml:25]: What is the difference with [GitHub release note generator](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)?

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 4 above)
Comment (JacquesDurand) [reply .github/workflows/release.yaml:18]: It is not the same kind of "release"

File: cog.toml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+path = "CHANGELOG.md"
+template = "remote"
+remote = "github.com"
+repository = "php-documentation-generator"
+owner = "php-documentation-generator"
+authors = [
+    { signature = "Hugo Nicolas", username = "JacquesDurand" },
+    { signature = "Antoine Bluchet", username = "soyuka" },
+    { signature = "Vincent Chalamon", username = "vincentchalamon" },
+]
```
Comment (vincentchalamon) [inline cog.toml]: EOF break line missing

**Pull request merged**